src/wallet/rpcdump.cpp

```diff
@@ -934,9 +934,9 @@ static std::string RecurseImportData(const CScript& script, ImportData& import_d
     case TxoutType::NONSTANDARD:
     case TxoutType::WITNESS_UNKNOWN:
     case TxoutType::WITNESS_V1_TAPROOT:
-    default:
         return "unrecognized script";
-    }
+    } // no default case, so the compiler can warn about missing cases
+    CHECK_NONFATAL(false);
```

Comment:
I understand that RPC code shouldn't terminate the server regardless of user's input. But `TxoutType` is an `enum class`, and its set of values does not depends on user's input. Therefore, I think the usual `assert(false);` should be used here.

Reply:
What is the difference in practise? This is dead code either way, so it doesn't matter what it does as long as it ensures that the function doesn't return normally when the dead code is executed (for whatever reason).

Reply:
The only difference is a uniform approach through the repo.

Reply:
The rpc code has a linter to check for a uniform `CHECK_NONFATAL` in all rpc code ;)